astropy/time/formats.py

```diff
@@ -183,10 +210,24 @@ def _check_val_type(self, val1, val2):
 
         if getattr(val1, 'unit', None) is not None:
             # Possibly scaled unit any quantity-likes should be converted to
-            _unit = u.CompositeUnit(getattr(self, 'unit', 1.), [u.day], [1])
-            val1 = u.Quantity(val1, copy=False).to_value(_unit)
+            val1 = u.Quantity(val1, copy=False)
             if val2 is not None:
-                val2 = u.Quantity(val2, copy=False).to_value(_unit)
+                val2 = u.Quantity(val2, copy=False)
+
+            try:
+                val1, val2 = careful_quantity_to_day(val1, val2)
+            except u.UnitsError:
+                raise u.UnitsError("only quantities with time units can be "
```

Comment:
Not clear if our testing is inadequate, but this looks like an API change: the exception type and message are both different.

Reply:
Ah, you are right. I don't think anyone should count on the exception message, but the exception type is bad. I changed it to `UnitConversionError`, which is than a `ValueError` subclass (but more accurate than just `ValueError`).